tools/physiofit/physiofit.xml

```diff
@@ -0,0 +1,240 @@
+                </param>
+                <when value='Yes'>
+                    <param name='flux_met_bounds' type='text' value='()' label='Bounds on metabolite fluxes. Bounds should be given in tuple format (see docs for more information)' />
+                </when>
+            </conditional>
+
+            <conditional name='cb_select'>
+                <param name='cb' type='select' label='Select yes to add bounds on biomass concentrations'>
+                    <option value='No'>No</option>
+                    <option value='Yes'>Yes</option>
+                </param>
+                <when value='Yes'>
+                    <param name='conc_biom_bounds' type='text' value='()' label='Bounds on biomass concentrations. Bounds should be given in tuple format (see docs for more information)' />
+                </when>
+            </conditional>
+
+            <conditional name='fb_select'>
+                <param name='fb' type='select' label='Select yes to add bounds on biomass fluxes'>
+                    <option value='No'>No</option>
+                    <option value='Yes'>Yes</option>
+                </param>
+                <when value='Yes'>
+                    <param name='flux_biom_bounds' type='text' value='()' label='Bounds on biomass fluxes. Bounds should be given in tuple format (see docs for more information)' />
+                </when>
+            </conditional>
+
+            <param name='verbose' type='boolean' argument='--verbose' truevalue='-v' falsevalue='' label='Flag to get debug information'/>
+
+        </section>
+
+    </inputs>
+
+    <outputs>
+        <data name='plots' label='Plots' format='pdf'/>
+        <data name='fluxes' label='Flux_results' format='tabular'/>
+        <data name='stats' label='Stat_results' format='tabular'/>
+        <data name='config' label='Config_file' format='json'/>
+    </outputs>
+
+    <tests>
+        <test>
+            <param name='input_selection|input_selector' value='tsv' />
+            <param name='input_selection|datafile' value='KEIO_ROBOT6_1.tsv' />
+            <section name='basic_settings'>
+                <param name='lag' value='false'/>
+                <param name='deg_select|deg' value='false'/>
+                <param name='mc_select|montecarlo' value='false'/>
+            </section>
+            <section name='advanced_settings'>
+                <param name='vini' value='1'/>
+                <param name='sd_select|sd' value='false'/>
+                <param name='cm_select|cm' value='false'/>
+                <param name='fm_select|fm' value='false'/>
+                <param name='cb_select|cb' value='false'/>
+                <param name='fb_select|fb' value='false'/>
+                <param name='verbose' value='false'/>
+            </section>
+            <output name='fluxes'>
+                <assert_contents>
+                    <has_n_columns n='7'/>
+                </assert_contents>
+            </output>
```

Comment:
Your test isn't really strong, isn't it?

Why don't you use `test-data/flux_results.tsv`?

Reply:
I had trouble using the flux_results.tsv file as comparison, I do not know why. I added a file size check and a line number check, it should be good enough now right?

Reply:
My guess is that the results have too many floats.
Values that are too precise may vary from machine to machine

Reply:
In the case of floats imprecision, you can check the existence of some lines that matches regex:
https://docs.galaxyproject.org/en/latest/dev/schema.html#id141

for example:
```xml
                <assert_contents>
                    <has_text_matching expression="X_0\s0\.0198\d+\s0\.02\d+\s0\.0019\d+\s0\.02\d+\s0\.016\d+\s0\.02399\d+" />
                </assert_contents>
```
You can try to match each value (like I did), or you can match only some specific values - just to check you actually have values. You can even just replace whole numbers by regex to check there are no "NA" or "NULL" or missing values:
```X_0(\s\d+\.\d+){6}```

Reply:
As the doc says: «  If the regular expression is expected to match a particular number of times, this value can be specified using n. »
```
                <assert_contents>
                    <has_text_matching expression="[a-zA-Z0-9_]+(\s\-?\d+(e\-?|\.)\d+){6}" n="6" />
                </assert_contents>
```
With this you can check you have six lines, each beginning with at least one or more alphanumeric character or underscores, followed by the pattern: a space character (tabs are considered as spaces characters), followed by a number,  six times.